appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/utils/PropertiesUtil.java

```diff
@@ -131,9 +131,15 @@ public static void populateProperties(MockComponent mockComponent, List<Componen
     String componentType = mockComponent.getType();
     // Configure properties
     for (ComponentDatabaseInterface.PropertyDefinition property : propertyDefinitions) {
+      String propertyDesc = ComponentsTranslation.getPropertyDescription(property.getName()
+          + "PropertyDescriptions");
+      if (propertyDesc.equals(property.getName() + "PropertyDescriptions")) {
+        propertyDesc = ComponentsTranslation.getPropertyDescription((componentType.equals("Form")
+             ? "Screen" : componentType) + "." + propertyDesc);
+      }
       mockComponent.addProperty(property.getName(), property.getDefaultValue(),
           ComponentsTranslation.getPropertyName(property.getCaption()),
-          property.getCategory(), property.getDescription(),
+          ComponentsTranslation.getCategoryName(property.getCategory()), propertyDesc,
```

Comment:
Because of how the multiselect functionality works, we might want to move the i18n logic into the property help widget and the property panel. Otherwise, we have to change every location we add properties, etc.

Reply:
> Because of how the multiselect functionality works, we might want to move the i18n logic into the property help widget and the property panel. Otherwise, we have to change every location we add properties, etc.

The reason this logic is here is because we don't currently have access to the componentType in the help widget or the property panel. This isn't necessarily where I think the logic should go, just a starting place in the conversation. This is a place where it works without changing class attributes or constructors.

Reply:
Hmm. So I have been testing this a bit. In particular, I was using Catalan as the testing language. In some cases, we directly add properties rather than going through this particular code path. This results in properties being further segmented since those other code paths aren't handling the i18n like the changes here. In particular, if I open HelloPurr and select the button, the Width/Height properties still have the category untranslated as Appearance whereas the other properties are translated.

I think what we should do is change the logic in MockComponent#addProperty (called on this line) so that it performs the i18n and therefore multiple code paths are covered. At that point you will still have access to both the component type and the property name and you can perform the lookup rather than doing it here.

If we make that change and it works, I would recommend merging this for nb196.

Reply:
For example, the call in MockVisibleComponent#addWidthHeightProperties passes the caption without translating it, which leads to the effect I saw in the category names.

Reply:
> I think what we should do is change the logic in MockComponent#addProperty (called on this line) so that it performs the i18n and therefore multiple code paths are covered. At that point you will still have access to both the component type and the property name and you can perform the lookup rather than doing it here.

This sounds good. I'll look at it.